scripts/platforms/summit/spack.yaml

```diff
@@ -0,0 +1,84 @@
+spack:
+  specs:
+  - hiop@develop+cuda+raja+sparse+mpi+deepchecking+ginkgo+cusolver cuda_arch=70 build_type=Release
```

Comment:
@cnpetra Do you want to activate `deepchecking` in the `Release` mode?

Reply:
makes no sense

Reply:
@nkoukpaizan @CameronRutherford let's remove `deepchecking` when `build_type=Release`

Reply:
Deep checking was added for builds used on CI pipelines. Somehow that slipped into other builds.

Reply:
We shoud take this into consideration when merging https://github.com/spack/spack/pull/32750 @ryandanehy